src/Core/Utilities/CaptchaProtectedAttribute.cs

```diff
@@ -0,0 +1,37 @@
+using Bit.Core.Context;
+using Bit.Core.Exceptions;
+using Bit.Core.Services;
+using Bit.Core.Models.Api;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Bit.Core.Utilities
+{
+    public class CaptchaProtectedAttribute : ActionFilterAttribute
+    {
+        public string ModelParameterName { get; set; } = "model";
+
+        public override void OnActionExecuting(ActionExecutingContext context)
+        {
+            var currentContext = context.HttpContext.RequestServices.GetRequiredService<ICurrentContext>();
+            var captchaValidationService = context.HttpContext.RequestServices.GetRequiredService<ICaptchaValidationService>();
+
+            if (captchaValidationService.ServiceEnabled && (currentContext.IsBot || captchaValidationService.RequireCaptcha))
```

Comment:
Would there be a scenario where `IsBot != RequireCaptcha`? It seems like this particular logic should be fully encapsulated inside of `ICaptchaValidationService` unless I'm mistaken.

Reply:
That's a good point. especially since there's multiple locations now

Reply:
BTW, the require captcha is probably better named as _force_ captcha. The idea being that we can force captcha on all requests. Primarily I wrote that for QA, but it could be useful if we find ourselves struggling under some future attack.

Reply:
Good idea on the rename to "ForceCaptcha"